controller/rancher/certificate_fetcher.go

```diff
@@ -263,6 +263,10 @@ func (fetcher *RCertificateFetcher) LookForCertUpdates(doOnUpdate func(string))
 }
 
 func (fetcher *RCertificateFetcher) readCertificate(path string, f os.FileInfo, err error) error {
+	if err != nil {
+		logrus.Debugf("Error %v while walking dir %v", err, path)
+		return err
```

Comment:
can we return it in fmt.errorf format? so you know that the error happened while walking which dir